emailer.py

```diff
@@ -0,0 +1,167 @@
+            return None
+
+        if len(integrations) > 1:
+            # If there are multiple integrations configured, none of
+            # them can be the 'site-wide' configuration.
+            raise CannotLoadConfiguration(
+                'Multiple email integrations are configured'
+            )
+
+        [integration] = integrations
+        return integration
+
+    def __init__(self, smtp_username, smtp_password, smtp_host, smtp_port,
+                 from_address, templates):
+        """Constructor."""
+        if not smtp_username:
+            raise CannotLoadConfiguration("No SMTP username specified")
+        self.smtp_username = smtp_username
+        if not smtp_password:
+            raise CannotLoadConfiguration("No SMTP password specified")
+        self.smtp_password = smtp_password
+        if not smtp_host:
+            raise CannotLoadConfiguration("No SMTP host specified")
+        self.smtp_host = smtp_host
+        if not smtp_host:
+            raise CannotLoadConfiguration("No SMTP port specified")
+        self.smtp_port = smtp_port
+        if not from_address:
+            raise CannotLoadConfiguration("No From: address specified")
+        self.from_address = from_address
+        self.templates = templates
+
+    def send(self, email_type, to_address, smtp=None, **kwargs):
+        """Generate an email from a template and send it.
+
+        :param email_type: The name of the template to use.
+        :param to_address: Addressee of the email.
+        :param smtp: Use this object as a mock instead of creating an
+            smtplib.SMTP object.
+        :param kwargs: Arguments to use when generating the email from
+            a template.
+        """
+        if not email_type in self.templates:
+            raise ValueError("No such email template: %s" % email_type)
+        template = self.templates[email_type]
+        body = template.body(self.from_address, to_address, **kwargs)
+        return self._send_email(to_address, body, smtp)
+
+    def _send_email(self, to_address, body, smtp=None):
+        """Actually send an email."""
+        smtp = smtp or smtplib.SMTP()
+        smtp.connect(self.smtp_host, self.smtp_port)
+        smtp.login(self.smtp_username, self.smtp_password)
+        smtp.sendmail(self.from_address, to_address, body)
+        smtp.quit()
+
+
+class EmailTemplate(object):
+    """A template for email messages."""
+
+    def __init__(self, subject_template, body_template):
+        self.subject_template = subject_template
+        self.body_template = body_template
+
+    def body(self, from_address, to_address, **kwargs):
+        """Generate the complete body of the email message, including headers.
+
+        :param from_address: Originating address.
+        :param to_address: Destination address.
+        :param kwargs: Arguments to use when filling out the template.
+        """
+
+        message = email.Message.Message()
+        message['From'] = from_address
```

Comment:
should this be something like "Library Simplified Library Registry" instead of the email address?